internal/v4/archive.go

```diff
@@ -162,7 +175,21 @@ func (h *Handler) servePutArchive(id *charm.Reference, w http.ResponseWriter, re
 	if req.ContentLength == -1 {
 		return badRequestf(nil, "Content-Length not specified")
 	}
-	if err := h.addBlobAndEntity(id, req.Body, hash, req.ContentLength); err != nil {
+	if err := req.ParseForm(); err != nil {
```

Comment:
It turns out we do, also we were using the result higher up in the function. Redundant code removed.